vm/compiler/src/ledger/block/bytes.rs

```diff
@@ -32,10 +32,22 @@ impl<N: Network> FromBytes for Block<N> {
         let previous_hash = FromBytes::read_le(&mut reader)?;
         let header = FromBytes::read_le(&mut reader)?;
         let transactions = FromBytes::read_le(&mut reader)?;
+
+        // Write the coinbase proof.
+        let coinbase_variant = u8::read_le(&mut reader)?;
+        let coinbase_proof = match coinbase_variant {
+            0 => None,
+            1 => Some(FromBytes::read_le(&mut reader)?),
+            _ => return Err(error("Invalid coinbase proof variant")),
+        };
```

Comment:
it might make sense to introduce an
```
#[repr(u8)]
enum CoinbaseVariant {
    Zero,
    One,
}
```
which implements `TryFrom<u8>`

Reply:
We can even be more explicit:
```rust
#[repr(u8)]
enum CoinbaseVariant {
	Zero = 0,
	One = 1,
}
```